zhaquirks/danfoss/thermostat.py

```diff
@@ -26,113 +54,350 @@
     OUTPUT_CLUSTERS,
     PROFILE_ID,
 )
-from zhaquirks.danfoss import D5X84YU, DANFOSS
-
-
-class DanfossThermostatCluster(CustomCluster, Thermostat):
-    """Danfoss custom cluster."""
-
-    server_commands = Thermostat.server_commands.copy()
-    server_commands[0x40] = foundation.ZCLCommandDef(
-        "setpoint_command",
-        {"param1": t.enum8, "param2": t.int16s},
-        is_manufacturer_specific=True,
-    )
-
-    attributes = Thermostat.attributes.copy()
-    attributes.update(
-        {
-            0x4000: ("etrv_open_windows_detection", t.enum8, True),
-            0x4003: ("external_open_windows_detected", t.Bool, True),
-            0x4010: ("exercise_day_of_week", t.enum8, True),
-            0x4011: ("exercise_trigger_time", t.uint16_t, True),
-            0x4012: ("mounting_mode_active", t.Bool, True),
-            0x4013: ("mounting_mode_control", t.Bool, True),
-            0x4014: ("orientation", t.Bool, True),
-            0x4015: ("external_measured_room_sensor", t.int16s, True),
-            0x4016: ("radiator_covered", t.Bool, True),
-            0x4020: ("control_algorithm_scale_factor", t.uint8_t, True),
-            0x4030: ("heat_available", t.Bool, True),
-            0x4031: ("heat_supply_request", t.Bool, True),
-            0x4032: ("load_balancing_enable", t.Bool, True),
-            0x4040: ("load_radiator_room_mean", t.uint16_t, True),
-            0x404A: ("load_estimate_radiator", t.uint16_t, True),
-            0x404B: ("regulation_setPoint_offset", t.int8s, True),
-            0x404C: ("adaptation_run_control", t.enum8, True),
-            0x404D: ("adaptation_run_status", t.bitmap8, True),
-            0x404E: ("adaptation_run_settings", t.bitmap8, True),
-            0x404F: ("preheat_status", t.Bool, True),
-            0x4050: ("preheat_time", t.uint32_t, True),
-            0x4051: ("window_open_feature_on_off", t.Bool, True),
-            0xFFFD: ("cluster_revision", t.uint16_t, True),
-        }
-    )
+from zhaquirks.quirk_ids import DANFOSS_ALLY_THERMOSTAT
+
+DANFOSS = "Danfoss"
+HIVE = DANFOSS
+POPP = "D5X84YU"
+
+OCCUPIED_HEATING_SETPOINT_NAME = "occupied_heating_setpoint"
+SYSTEM_MODE_NAME = "system_mode"
```

Comment:
For getting ZCL attribute ids or names, you can use: `Thermostat.AttributeDefs.system_mode.name`
(or `.id` for the id)

Reply:
Thanks, that tidies the constants up nicely.